### PR TITLE
Unify icons, remove png in favor of xml

### LINKED ISSUE #3783: Icon inconsistency
The app uses both outline and filled icons:
image

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
- removed png files in favor of xml
- added corresponding svg to drawable_ressource
- Resolves #3783, replaces the outlines folder icon in the drawer menu with a filled one

**Open Issues**

- [x] mutate icons where necessary (no more issues found)
- [x] test this on Android 4
- [x]  settings icon in drawer menu seems to be the gear icon of Nc instead of the Android gear icon
- [x] avatars in userlist get tinted but shouldn't

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
:-1: needs further work due to the tinting issues...

* all files -> open folder details for a faved folder -> navigate back -> open notifications via drawer menu -> open all files -> open drawer -> the favorites menu item has a yellow icon... (instead of just grey)

* open favorites in list view -> all share icons (the share not the share via link icon) are grey not matter the share-state, so tinting isn't working here...

@mario @tobiasKaminsky I guess this will happen in even more places and I think @mario had this issue with talk (too?) and had a fix for this? :)

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky so you suggest to make the drawable mutable or would you rather duplicate the drawable for different, fixed colors so tinting won't be an issue?

**Comment by tobiasKaminsky** [comment]
Mutating should be fine, otherwise we end up with multiple drawables…

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger have you done the mutating part? Otherwise i can have a look next week

**Comment by AndyScherzinger** [comment]
> have you done the mutating part? Otherwise i can have a look next week

@tobiasKaminsky I didn't find the time, so if you find the time next week, that be great!

**Comment by mario** [comment]
@tobiasKaminsky this PR doesn't seem finished? :)

**Comment by AndyScherzinger** [comment]
@mario correct, the mutation part is missing!

**Comment by tobiasKaminsky** [comment]
Rebased.
@AndyScherzinger https://github.com/nextcloud/android/blob/80542bbff7ef19f5dc37fd80471f4fa0d507607f/src/main/java/com/owncloud/android/ui/adapter/OCFileListAdapter.java#L849-L850 this is the mutable part

**Comment by AndyScherzinger** [comment]
The favorite/star icon seems to work now (at least for me). The share icon still has the problem while this is probably not related to this PR (not sure):

open favorites in list view -> all share icons (the share not the share via link icon) are grey not matter the share-state, so tinting isn't working here...

All entries show the grey share icon.
To reproduce: have 3 folders which are favorites where:
1. not shared at all
2. shared via link
3. shared to a user

The share state is correct in the "all files" view while the "favorites" vie shows the grey share icon for all 3 entries instead of 1. grey share, 2. black link icon, 3. black share icon

**Comment by tobiasKaminsky** [comment]
As we will have RC1 of 3.6.0 next week, I suggest that we try to get this done & ready for the week after.
So that we have one complete dev cycle to test it with dev version (and of course see it in daily development)

I'll have a look into it tomorrow.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky sounds good to me. Unfortunattely I didn't find the time to look into all the necessary mutate calls.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky I did yet another rebase + fixing the missing sign-offs for my commits. ❗️

**Comment by tobiasKaminsky** [comment]
Disabled:
VectorPath: 
> Using long vector paths is bad for performance. There are several ways to make the pathData shorter:
>* Using less precision
>* Removing some minor details
>* Using the Android Studio vector conversion tool
>* Rasterizing the image (converting to PNG)

We use Android Studio to generate xml, so we have no influence.

IconMissingDensityFolder:
> Missing density variation folders in src/main/res: drawable-mdpi, drawable-xhdpi
Icons will look best if a custom version is provided for each of the major screen density classes (low, medium, high, extra-high, extra-extra-high). This lint check identifies folders which are missing, such as drawable-hdpi.

We do not have those folders as all icons are done via xml.

IconDensities:
> ../../../src/main/res/drawable-hdpi: Missing the following drawables in drawable-hdpi: apk.png, background.jpg, background_nc13.png, fdroid.png, owncloud_progressbar_indeterminate_1.png... (8 more)
../../../src/main/res/drawable-xxhdpi: Missing the following drawables in drawable-xxhdpi: apk.png, background.jpg, background_nc13.png, fdroid.png, logo.png... (2 more)

These are images, we only have/need in one resolution.

**Comment by tobiasKaminsky** [comment]
Increasing FindBug is due to:
> UR_UNINIT_READ: Uninitialized read of field in constructor
This constructor reads a field which has not yet been assigned a value.  This is often caused when the programmer mistakenly uses the field instead of one of the constructor's parameters.

https://github.com/nextcloud/android/blob/c7d2517873f5cff590cfae03dac12754f499128e/src/main/java/com/owncloud/android/ui/adapter/OCFileListAdapter.java#L953-L957

FindBug does not recognize that favorite is assigned via butterKnife.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky see screenshot, the issues have been added to the description at the top

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky I fixed all remaining issues 🎉 Only thing I can't do are Android 4 tests since I can't get an emulator to run an Android 4 image :/

❗️ rebased the branch ❗️

**Comment by tobiasKaminsky** [comment]
Android 4.1:
```
03-27 16:30:32.393 2080-2080/com.nextcloud.client E/AndroidRuntime: FATAL EXCEPTION: main
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.nextcloud.client/com.owncloud.android.ui.activity.FileDisplayActivity}: android.view.InflateException: Binary XML file line #31: Error inflating class TextView
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2059)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2084)
        at android.app.ActivityThread.access$600(ActivityThread.java:130)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1195)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:137)
        at android.app.ActivityThread.main(ActivityThread.java:4745)
        at java.lang.reflect.Method.invokeNative(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:511)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
        at dalvik.system.NativeStart.main(Native Method)
     Caused by: android.view.InflateException: Binary XML file line #31: Error inflating class TextView
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:704)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:746)
        at android.view.LayoutInflater.parseInclude(LayoutInflater.java:830)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:736)
        at android.view.LayoutInflater.parseInclude(LayoutInflater.java:830)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:736)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:749)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:489)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:396)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:352)
        at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:469)
        at androidx.appcompat.app.AppCompatActivity.setContentView(AppCompatActivity.java:140)
        at com.owncloud.android.ui.activity.FileDisplayActivity.onCreate(FileDisplayActivity.java:239)
        at android.app.Activity.performCreate(Activity.java:5008)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1079)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2023)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2084) 
        at android.app.ActivityThread.access$600(ActivityThread.java:130) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1195) 
        at android.os.Handler.dispatchMessage(Handler.java:99) 
        at android.os.Looper.loop(Looper.java:137) 
        at android.app.ActivityThread.main(ActivityThread.java:4745) 
        at java.lang.reflect.Method.invokeNative(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:511) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553) 
        at dalvik.system.NativeStart.main(Native Method) 
     Caused by: android.content.res.Resources$NotFoundException: File res/drawable/ic_information_outline.xml from drawable resource ID #0x7f0800af
        at android.content.res.Resources.loadDrawable(Resources.java:1923)
        at android.content.res.TypedArray.getDrawable(TypedArray.java:601)
        at android.widget.TextView.<init>(TextView.java:614)
        at androidx.appcompat.widget.AppCompatTextView.<init>(AppCompatTextView.java:87)
        at androidx.appcompat.widget.AppCompatTextView.<init>(AppCompatTextView.java:83)
        at androidx.appcompat.app.AppCompatViewInflater.createTextView(AppCompatViewInflater.java:177)
        at androidx.appcompat.app.AppCompatViewInflater.createView(AppCompatViewInflater.java:102)
        at androidx.appcompat.app.AppCompatDelegateImpl.createView(AppCompatDelegateImpl.java:1266)
        at androidx.appcompat.app.AppCompatDelegateImpl.onCreateView(AppCompatDelegateImpl.java:1316)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:675)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:746) 
        at android.view.LayoutInflater.parseInclude(LayoutInflater.java:830) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:736) 
        at android.view.LayoutInflater.parseInclude(LayoutInflater.java:830) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:736) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:749) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:489) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:396) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:352) 
        at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:469) 
        at androidx.appcompat.app.AppCompatActivity.setContentView(AppCompatActivity.java:140) 
        at com.owncloud.android.ui.activity.FileDisplayActivity.onCreate(FileDisplayActivity.java:239) 
        at android.app.Activity.performCreate(Activity.java:5008) 
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1079) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2023) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2084) 
        at android.app.ActivityThread.access$600(ActivityThread.java:130) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1195) 
        at android.os.Handler.dispatchMessage(Handler.java:99) 
        at android.os.Looper.loop(Looper.java:137) 
        at android.app.ActivityThread.main(ActivityThread.java:4745) 
        at java.lang.reflect.Method.invokeNative(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:511) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553) 
        at dalvik.system.NativeStart.main(Native Method) 
     Caused by: org.xmlpull.v1.XmlPullParserException: Binary XML file line #18: invalid drawable tag vector
        at android.graphics.drawable.Drawable.createFromXmlInner(Drawable.java:877)
        at android.graphics.drawable.Drawable.createFromXml(Drawable.java:818)
        at android.content.res.Resources.loadDrawable(Resources.java:1920)
        at android.content.res.TypedArray.getDrawable(TypedArray.java:601) 
        at android.widget.TextView.<init>(TextView.java:614) 
        at androidx.appcompat.widget.AppCompatTextView.<init>(AppCompatTextView.java:87) 
        at androidx.appcompat.widget.AppCompatTextView.<init>(AppCompatTextView.java:83) 
        at androidx.appcompat.app.AppCompatViewInflater.createTextView(AppCompatViewInflater.java:177) 
        at androidx.appcompat.app.AppCompatViewInflater.createView(AppCompatViewInflater.java:102) 
        at androidx.appcompat.app.AppCompatDelegateImpl.createView(AppCompatDelegateImpl.java:1266) 
        at androidx.appcompat.app.AppCompatDelegateImpl.onCreateView(AppCompatDelegateImpl.java:1316) 
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:675) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:746) 
        at android.view.LayoutInflater.parseInclude(LayoutInflater.java:830) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:736) 
        at android.view.LayoutInflater.parseInclude(LayoutInflater.java:830) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:736) 
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:749) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:489) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:396) 
        at android.view.LayoutInflater.inflate(LayoutInflater.java:352) 
        at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:469) 
        at androidx.appcompat.app.AppCompatActivity.setContentView(AppCompatActivity.java:140) 
        at com.owncloud.android.ui.activity.FileDisplayActivity.onCreate(FileDisplayActivity.java:239) 
        at android.app.Activity.performCreate(Activity.java:5008) 
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1079) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2023) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2084) 
        at android.app.ActivityThread.access$600(ActivityThread.java:130) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1195) 
        at android.os.Handler.dispatchMessage(Handler.java:99) 
        at android.os.Looper.loop(Looper.java:137) 
        at android.app.ActivityThread.main(ActivityThread.java:4745) 
        at java.lang.reflect.Method.invokeNative(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:511) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553) 
        at dalvik.system.NativeStart.main(Native Method) 
```

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky thanks for the test, https://github.com/nextcloud/android/pull/2796/commits/96a9399eb2e955c35127a09668e92a5bfdc3f616 might fix this (also to be tested for the login screen!). I say "should" because I can't test on Android 4 (emulator deson't work properly on my laptop...)

Could you maybe give it another test drive? ❤️

File: src/main/java/com/owncloud/android/authentication/AuthenticatorActivity.java [hunk 1]
```diff
@@ -627,6 +631,12 @@ private void initOverallUi() {
         mServerStatusView = findViewById(R.id.server_status_text);
         mTestServerButton = findViewById(R.id.testServerButton);
 
+        // compound drawables need to be set programmatically (minSDK would need to be lollipop to use XML)
+        mServerStatusView.setCompoundDrawablesWithIntrinsicBounds(
+            (AppCompatResources.getDrawable(this, android.R.drawable.stat_notify_sync)), null, null, null);
+        mAuthStatusView.setCompoundDrawablesWithIntrinsicBounds(
+            (AppCompatResources.getDrawable(this, android.R.drawable.stat_notify_sync)), null, null, null);
```
[the remark this answers is not in the document]
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/authentication/AuthenticatorActivity.java]: fixed

**Comment by tobiasKaminsky** [comment]
https://github.com/nextcloud/android/blob/169c48c171341096dc0333290514e25ccdfe3020/build.gradle#L96

When disabling this line it works fine for me with android 4.1.

**Comment by tobiasKaminsky** [comment]
Testing with 4.0.4 on FirstRun the talk icon looks blurry

**Comment by AndyScherzinger** [comment]
> Testing with 4.0.4 on FirstRun the talk icon looks blurry

No it gets tricky ;) _If_ The vector icon does not have the right dp-sizing (as in the size it is normally used in), then the generated icons will follow that sizing in pixels and the UI will then have to resize them... This doesn't show up for vectors of course since they scale without getting blurry...

**Comment by tobiasKaminsky** [comment]
But the other items on FirstRun look fine…Can't we use the dp of them?

**Comment by AndyScherzinger** [comment]
> But the other items on FirstRun look fine…Can't we use the dp of them?

If it isn't used anywheree else in a different size then I think so. If it is used in different places in different sizes we either need to pick the largest version (since downscaling isn't as bad as upscaling) or we need to create an xml drawable for all the different sizes in use to have the best possible png versions for each scenario.

**Comment by tobiasKaminsky** [comment]
Tested on 4.0.4, 4.1 and 9.0.
Ready for squash (only this one please because of those many changes, which will then make rebase a lot easier for existing PRs) and then we can merge it.

**Comment by AndyScherzinger** [comment]
🎉 finally done - nice collab as always @tobiasKaminsky ❤️

**Pull request merged**